doc/tools.md

```diff
@@ -234,3 +234,117 @@ Below are some examples on how to use the `sendtx` tool in practice.
+
+    ./spvnode -b scan
+
+To utilize checkpoints for faster initial sync, apply the -p flag:
+
+    ./spvnode -p scan
+
+| Flag | Name | Required Arg? | Usage |
+|------|------|---------------|-------|
+| `-t`, `--testnet` | Testnet Mode | No | Activate testnet: `./spvnode -t scan` |
+| `-r`, `--regtest` | Regtest Mode | No | Activate regtest network: `./spvnode -r scan` |
+| `-i`, `--ips` | Initial Peers | Yes | Specify initial peers: `./spvnode -i 127.0.0.1:22556 scan` |
+| `-d`, `--debug` | Debug Mode | No | Enable debug output: `./spvnode -d scan` |
+| `-m`, `--maxnodes` | Max Peers | No | Set max peers: `./spvnode -m 8 scan` |
+| `-a`, `--address` | Address | Yes | Use address: `./spvnode -a "your address here" scan` |
+| `-n`, `--mnemonic` | Mnemonic Seed | Yes | Use BIP39 mnemonic: `./spvnode -n "your mnemonic here" scan` |
+| `-s`, `--pass_phrase` | Passphrase | Yes | Passphrase for BIP39 seed: `./spvnode -s "your passphrase" scan` |
+| `-f`, `--dbfile` | Database File | No | Headers DB file: `./spvnode -f ./headers.db scan` |
+| `-c`, `--continuous` | Continuous Mode | No | Run continuously: `./spvnode -c scan` |
+| `-b`, `--full_sync` | Full Sync | No | Perform a full sync: `./spvnode -b scan` |
+| `-p`, `--checkpoint` | Checkpoint | No | Enable checkpoint sync: `./spvnode -p scan` |
+| `-w`, `--wallet_file` | Wallet File | Yes | Specify wallet file: `./spvnode -w "./wallets/wallet.db" scan` |
+| `-h`, `--headers_file` | Headers File | Yes | Specify headers DB file: `./spvnode -h "./headers/headers.db" scan` |
+| `-y`, `--encrypted_file` | Encrypted File | Yes | Use encrypted file: `./spvnode -y 0 scan` |
+| `-j`, `--use_tpm` | Use TPM | No | Utilize TPM for decryption: `./spvnode -j scan` |
+| `-k`, `--master_key` | Master Key | No | Use master key decryption: `./spvnode -k scan` |
+| `-z`, `--daemon` | Daemon Mode | No | Run as a daemon: `./spvnode -z scan` |
+
+### Commands
+
+The primary command for `spvnode` is `scan`, which syncs the blockchain headers:
+
+#### `scan`
+Connects to the Dogecoin network and synchronizes the blockchain headers to the local database.
+
+### Callback Functions
+
+The tool provides several callbacks for custom integration:
+
+- `spv_header_message_processed`: Triggered when a header is processed.
+- `spv_sync_completed`: Invoked upon completion of the sync process.
+
+### Best Practices and Notes
+When not specifying -w, spvnode will default to using main_wallet.db. To prevent unintended interactions with main_wallet.db, it's important to be consistent with the use of flags. The best practice is to always use -w and specify a distinct wallet file, especially when using new mnemonics or keys.
+
+When using -n with a mnemonic, instead of main_wallet.db, spvnode will generate main_mnemonic_wallet.db.
+
+## Examples
+
+#### Sync up to the chain tip and stores all headers in `headers.db` (quit once synced):
+    spvnode scan
```

Comment:
small nit with this and all the commands below as having to do with local vs installed. perhaps we should append ./ to ensure the binary being used is the local version to avoid confusion around the situation i encountered yesterday.